pxtsim/simdriver.ts

```diff
@@ -92,7 +92,9 @@ namespace pxsim {
                 const simUrl = new URL(this.getSimUrl())
                 this._allowedOrigins.push(simUrl.origin)
             } catch (e) {
-                console.error(`Invalid sim url ${this.getSimUrl()}`)
+                if (!U.isLocalHost()) {
```

Comment:
This looks fine to me. We really only want to log an error when the sim URL is invalid when we're not running on localhost. If the sim URL were invalid in that case, then the simulator iframe would not load correctly and this error message would make it clear that the malformed URL is the culprit. 

Something we can consider is changing
```ts
private getSimUrl(): string {
    return this.options.simUrl || ((window as any).pxtConfig || {}).simUrl || "/sim/simulator.html"
}
```
to
```ts
private getSimUrl(): string {
    return this.options.simUrl || ((window as any).pxtConfig || {}).simUrl || location.origin + "/sim/simulator.html"
}
```
This way we'd have a valid URL even when we're on localhost. When we're creating the simulator iframe:
```ts
frame.src = this.getSimUrl() + '#' + frame.id;
```
in the 
```ts
private createFrame(light?: boolean): HTMLDivElement {
```
function, setting the frame src to "/sim/simulator.html" + '#' + frame.id will actually include the origin of the window creating the frame (localhost in this case). So including the origin ourselves in the localhost case means we don't have to check for local host before logging an error about an invalid URL. It's probably what I should have done when I first created this PR. 

Either way, looks good to me.

Reply:
Good call, that sounds better to me too~